### PR TITLE
Fix a problem with rvc

### PR DESCRIPTION
**Pull request by RHL120**
**Checklist**

- [ ] Closing issues: #issue
- [X] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

rvc_git_commit was not working now it does.

### CODE REVIEW

File: libr/core/rvc.c [hunk 1]
```diff
@@ -733,30 +733,40 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
...
 					free (path);
 					free (editor);
 					return false;
-				} else {
-					free (editor);
-					free (path);
-					return false;
 				}
+			} else {
+				eprintf ("Please set the $EDITOR env var\n");
```
Comment (trufae) [inline libr/core/rvc.c:749]: Better inform the user about cfg.editor r2 var for portability

Comment (trufae) [reply libr/core/rvc.c:749]: Well actually r2 have a line editor like ed in rcons and this should be the spawned editor if none is set

Comment (RHL120) [reply libr/core/rvc.c:749]: how can I spawn it?

Comment (trufae) [reply libr/core/rvc.c:749]: seems like it wont work if cbeditor is set. ill fix in few mins. hold on

File: libr/core/rvc.c [hunk 2]
```diff
@@ -733,30 +734,48 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
...
 					free (editor);
 					return false;
-				} else {
-					free (editor);
+				}
+			} else {
+				free (editor);
+				free (path);
+				r_line_set_prompt ("Enter a commit message:");
+				r_core_visual_showcursor (NULL, true);
```
Comment (trufae) [inline libr/core/rvc.c:754]:
```suggestion
```

File: libr/core/rvc.c [hunk 3]
```diff
@@ -733,30 +734,48 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
...
+				r_cons_fgets (m, MAX_MESSAGE_LEN, 0, NULL);
+				r_core_visual_showcursor (NULL, false);
```
Comment (trufae) [inline libr/core/rvc.c:756]:
```suggestion
```

File: libr/core/rvc.c [hunk 4]
```diff
@@ -733,30 +734,48 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
...
 					free (path);
 					free (editor);
 					return false;
-				} else {
-					free (editor);
+				}
+			} else {
+				free (editor);
+				free (path);
+				r_line_set_prompt ("Enter a commit message:");
```
Comment (trufae) [inline libr/core/rvc.c:753]:
```suggestion
				r_line_set_prompt ("Commit message:");
```

File: libr/core/rvc.c [hunk 5]
```diff
@@ -733,30 +734,48 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
...
-	if (!message) {
+	if (R_STR_ISEMPTY (message)) {
 		char *path = NULL;
 		(void)r_file_mkstemp ("rvc", &path);
 		if (path) {
+			char m[MAX_MESSAGE_LEN + 1];
 			char *editor = r_sys_getenv ("EDITOR");
 			if (!R_STR_ISEMPTY (editor)) {
-				if (!r_sys_cmdf ("%s %s", editor, path)) {
+				if (r_sys_cmdf ("%s %s", editor, path)) {
```
Comment (trufae) [inline libr/core/rvc.c:744]: Use `r_cons_editor()` instead of introducing this code injection vuln

File: libr/core/rvc.c [hunk 6]
```diff
@@ -733,30 +734,48 @@ R_API bool r_vc_commit(const char *rp, const char *message, const char *author,
 		eprintf ("Can't commit");
 		return false;
 	}
-	if (!message) {
+	if (R_STR_ISEMPTY (message)) {
 		char *path = NULL;
 		(void)r_file_mkstemp ("rvc", &path);
 		if (path) {
+			char m[MAX_MESSAGE_LEN + 1];
 			char *editor = r_sys_getenv ("EDITOR");
```
Comment (trufae) [inline libr/core/rvc.c:742]: EDITOR is honored by cfg.editor. which is then used by r_cons_editor(), dont check for this env var. as its a unix-specific thing an r2 have the cfg.editor abstraction

**Review by trufae — CHANGES_REQUESTED** [review]

**Review by trufae — APPROVED** [review]

**Pull request merged**